packages/aws-cdk-lib/aws-rds/lib/instance.ts

```diff
@@ -132,6 +133,49 @@ export interface DatabaseInstanceAttributes {
  * A new or imported database instance.
  */
 export abstract class DatabaseInstanceBase extends Resource implements IDatabaseInstance {
+  /**
+   * Lookup an existing DatabaseInstance using instanceIdentifier.
+   */
+  public static fromLookup(scope: Construct, id: string, options: DatabaseInstanceLookupOptions): IDatabaseInstance {
+    const response: {[key: string]: any} = ContextProvider.getValue(scope, {
+      provider: cxschema.ContextProvider.CC_API_PROVIDER,
+      props: {
+        typeName: 'AWS::RDS::DBInstance',
+        exactIdentifier: options.instanceIdentifier,
+        propertiesToReturn: [
+          'DBInstanceArn',
+          'Endpoint.Address',
+          'Endpoint.Port',
+          'DbiResourceId',
+          'DBSecurityGroups',
+        ],
+      } as cxschema.CcApiContextQuery,
+      dummyValue: {},
+    }).value;
+
+    const instance = response[options.instanceIdentifier];
+
+    // Get ISecurityGroup from securityGroupId
+    const securityGroups: ec2.ISecurityGroup[] = [];
+    const dbsg: [string] = instance.DBSecurityGroups;
+    dbsg.forEach(securityGroupId => {
+      const securityGroup = ec2.SecurityGroup.fromSecurityGroupId(
+        scope,
+        `LSG-${securityGroupId}`,
+        securityGroupId,
+      );
```

Comment:
We probably need to add an option to the properties object, and pass it here to `fromSecurityGroupId`, on whether the security groups are going to be mutable or immutable by default.

The difference is whether we will try to add rules to them. `false` makes more sense, but due to historical reasons the default is `true`... ☹️